File: mobile/lib/config/specialists.dart
/// Specialist types for referral appointment requests.
/// Sorted alphabetically by label. Mirrors web/src/config/specialists.ts.
const List<MapEntry<String, String>> specialistTypes = [
  MapEntry('allergist', 'Allergist'),
  MapEntry('cardiologist', 'Cardiologist'),
  MapEntry('chiropractor', 'Chiropractor'),
  MapEntry('dermatologist', 'Dermatologist'),
  MapEntry('endocrinologist', 'Endocrinologist'),
  MapEntry('gastroenterologist', 'Gastroenterologist'),
  MapEntry('hematologist', 'Hematologist'),
  MapEntry('infectious_disease', 'Infectious Disease Specialist'),
  MapEntry('nephrologist', 'Nephrologist'),
  MapEntry('neurologist', 'Neurologist'),
  MapEntry('oncologist', 'Oncologist'),
  MapEntry('ophthalmologist', 'Ophthalmologist'),
  MapEntry('orthopedist', 'Orthopedist'),
  MapEntry('otolaryngologist', 'Otolaryngologist (ENT)'),
  MapEntry('pain_management', 'Pain Management'),
  MapEntry('physical_therapist', 'Physical Therapist'),
  MapEntry('plastic_surgeon', 'Plastic Surgeon'),
  MapEntry('podiatrist', 'Podiatrist'),
  MapEntry('psychiatrist', 'Psychiatrist'),
  MapEntry('pulmonologist', 'Pulmonologist'),
  MapEntry('radiologist', 'Radiologist'),
  MapEntry('rheumatologist', 'Rheumatologist'),
  MapEntry('urologist', 'Urologist'),
  MapEntry('vascular_surgeon', 'Vascular Surgeon'),
];

/// Get display label for a specialist value.
String getSpecialistLabel(String value) {
  for (final entry in specialistTypes) {
    if (entry.key == value) return entry.value;
  }
  return value;
}

File: web/src/config/specialists.ts
/**
 * Specialist types for referral appointment requests.
 * Sorted alphabetically by label.
 */
export const SPECIALIST_TYPES: { value: string; label: string }[] = [
  { value: 'allergist', label: 'Allergist' },
  { value: 'cardiologist', label: 'Cardiologist' },
  { value: 'chiropractor', label: 'Chiropractor' },
  { value: 'dermatologist', label: 'Dermatologist' },
  { value: 'endocrinologist', label: 'Endocrinologist' },
  { value: 'gastroenterologist', label: 'Gastroenterologist' },
  { value: 'hematologist', label: 'Hematologist' },
  { value: 'infectious_disease', label: 'Infectious Disease Specialist' },
  { value: 'nephrologist', label: 'Nephrologist' },
  { value: 'neurologist', label: 'Neurologist' },
  { value: 'oncologist', label: 'Oncologist' },
  { value: 'ophthalmologist', label: 'Ophthalmologist' },
  { value: 'orthopedist', label: 'Orthopedist' },
  { value: 'otolaryngologist', label: 'Otolaryngologist (ENT)' },
  { value: 'pain_management', label: 'Pain Management' },
  { value: 'physical_therapist', label: 'Physical Therapist' },
  { value: 'plastic_surgeon', label: 'Plastic Surgeon' },
  { value: 'podiatrist', label: 'Podiatrist' },
  { value: 'psychiatrist', label: 'Psychiatrist' },
  { value: 'pulmonologist', label: 'Pulmonologist' },
  { value: 'radiologist', label: 'Radiologist' },
  { value: 'rheumatologist', label: 'Rheumatologist' },
  { value: 'urologist', label: 'Urologist' },
  { value: 'vascular_surgeon', label: 'Vascular Surgeon' },
];

/** Get display label for a specialist value */
export function getSpecialistLabel(value: string): string {
  return SPECIALIST_TYPES.find(s => s.value === value)?.label || value;
}
